spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-fabric8-istio-it/pom.xml

```diff
@@ -61,6 +61,7 @@
 				<configuration>
 					<imageName>docker.io/springcloud/${project.artifactId}:${project.version}</imageName>
 					<imageBuilder>paketobuildpacks/builder</imageBuilder>
+					<skip>${skip.build.image}</skip>
```

Comment:
I am skipping images builds for some steps, so need this one across every project that has image builds

Reply:
Because the image is already built in the cache?

Reply:
So first I do:

```
- name: build with skip tests and skip images
run: ./mvnw -T 1C -s .settings.xml clean install -U -DskipTests -Dskip.build.image=true
```
build entire project, but with `-DskipTests -Dskip.build.image=true`, thus no images anywhere yet.

then, I do:

```
- name: build controllers project
uses:  ./.github/workflows/composites/build-controllers-project
```

that does:

```
cd spring-cloud-kubernetes-controllers
.././mvnw -T 1C clean install
cd ..
```

As such, this builds the controllers images. 

Same thing in the next step for integration projects:

```
cd spring-cloud-kubernetes-integration-tests
# build the images, but dont run the tests
.././mvnw -T 1C clean install -DskipTests
 cd ..
```

As a result, we have all images built. Then I tar images:


```
        mkdir -p /tmp/docker/images
        TAG=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout)
        cd spring-cloud-kubernetes-integration-tests
        while read integ_test; do
          docker save -o /tmp/docker/images/${integ_test}.tar docker.io/springcloud/${integ_test}:$TAG
        done < <(mvn -Dexec.executable='echo' -Dexec.args='${project.artifactId}' exec:exec -q | grep -v 'spring-cloud-kubernetes-integration-tests')
        cd ..
```

This is a while loop, that in plain english does: "get all modules that are present in `spring-cloud-kubernetes-integration-tests`, but skip `spring-cloud-kubernetes-integration-tests` itself. Then for each entry tar it and save that tar in /tmp/docker/images". 

Only now, I can actually create a "cache per workflow" to share these images with other steps. 

---
So to answer your question (I hope it makes sense), I skip building images in some initial steps on purpose. If some plain test fails, it does not matter if I build the image or not - the build overall will fail, so why build it pre-maturely? Especially, since it takes so much time. So this skipping acts as an optimization and as an aggregate step, of building images for a subsequent step (cache per workflow).

Reply:
makes sense

Reply:
as soon as I added the README, build started failing - it turns out it passed until now, only by pure luck. trying to understand how to fix this in github actions